lib/modules/dosomething/dosomething_reportback/includes/ReportbackItemTransformer.php

```diff
@@ -128,4 +129,29 @@ private function removeUnauthorizedStatuses($statuses) {
     return $statuses;
   }
 
+  /**
+   * Remove Reportback Items with unauthorized statuses based on user permission.
+   *
+   * @param  array $data
+   * @return mixed
+   * @throws Exception
+   */
+  private function removeUnauthorizedResults($data) {
```

Comment:
It's different in where the data gets checked. The other one checks the statuses in the `$filters` passed to the query... this one is removing unauthorized items after the query has already run. It depends on whether it's an `index` or `retrieve` method which is why it has to be two different methods.